File: little-bee/controller/router/di.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_DI = "/config/di"
const URL_ROUTER_DI_BY_ID = "/config/di/:di-id"
const URL_ROUTER_DI_QUERY = "/config/di/query"
const URL_ROUTER_DI_ALL = "/config/di/all"

func DiRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_DI, config.CreateDi)
	r.GET(URL_ROUTER_DI_BY_ID, config.RetrieveDi)
	r.PUT(URL_ROUTER_DI, config.UpdateDi)
	r.DELETE(URL_ROUTER_DI_BY_ID, config.DeleteDi)
	r.PUT(URL_ROUTER_DI_BY_ID, config.DeleteDiWithFlag)
	return r
}

File: frontend/console/src/api/di.js
import Axios from "@/http";

export function createDi(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/di', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveDi(diId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/di/${ diId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateDi(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/di', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteDi(diId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/di/${ diId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteDiWithFlag(diId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/di/${ diId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllDis() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/di/all').then(res => {
            resolve(res)
        })
    })
}

export function queryDi (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/di/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}